editor-service/src/main/java/com/qreal/robots/common/auth/OAuth2UserDetailsLoaderImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.qreal.robots.common.auth;
+
+import com.qreal.robots.database.users.client.UserService;
+import com.qreal.robots.database.users.model.UserRole;
+import com.racquettrack.security.oauth.OAuth2UserDetailsLoader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * OAuth requests handler.
+ */
+@Component
+public class OAuth2UserDetailsLoaderImpl implements OAuth2UserDetailsLoader {
+
+    public static final String ROLE_USER = "ROLE_USER";
+
+    @Autowired
+    private UserService userService;
+
+    @Override
+    public UserDetails getUserByUserId(Object id) {
+        if (!userService.isUserExist((String) id)) {
```

Comment:
We already use here User from springframework. I haven't heard about qualified import in java.